components/eamxx/src/control/atmosphere_driver.cpp

```diff
@@ -232,6 +232,14 @@ void AtmosphereDriver::create_grids()
 
   m_atm_logger->debug("  [EAMxx] Grids created.");
 
+  // If TMS process is enabled, SHOC needs to know to request tms' surface drag coefficient
+  // as a required field during the set_grid() call below, but SHOC does not have knowledge
+  // of other processes. The driver needs propgate this information to SHOC.
+  if(m_atm_process_group->has_process("tms") &&
+     m_atm_process_group->has_process("shoc")) {
+    setup_shoc_tms_links();
+  }
```

Comment:
`SHOCMacrophysics::set_grids()` is the first place that needs to know if TMS is a process, so call here to set `apply_tms=true` is shoc params.